avocado/plugins/runner_nrunner.py

```diff
@@ -16,18 +16,96 @@
 NRunner based implementation of job compliant runner
 """
 
+import asyncio
 import json
+import multiprocessing
 import os
-import time
+import random
 from copy import copy
 
 from avocado.core import nrunner
+from avocado.core.dispatcher import SpawnerDispatcher
+from avocado.core.plugin_interfaces import CLI, Init
 from avocado.core.plugin_interfaces import Runner as RunnerInterface
+from avocado.core.settings import settings
 from avocado.core.status.repo import StatusRepo
+from avocado.core.status.server import StatusServer
 from avocado.core.task.runtime import RuntimeTask
+from avocado.core.task.statemachine import TaskStateMachine, Worker
 from avocado.core.test_id import TestID
 
 
+class RunnerInit(Init):
+
+    name = 'nrunner'
+    description = '*EXPERIMENTAL* nrunner initialization'
+
+    def initialize(self):
+        section = 'nrunner'
+        help_msg = 'Shuffle the tasks to be executed'
+        settings.register_option(section=section,
+                                 key='shuffle',
+                                 default=False,
+                                 help_msg=help_msg,
+                                 key_type=bool)
+
+        help_msg = 'URI for the status server, usually a "HOST:PORT" string'
+        settings.register_option(section=section,
+                                 key='status_server_uri',
+                                 default='127.0.0.1:8888',
+                                 metavar="HOST:PORT",
+                                 help_msg=help_msg)
+
+        help_msg = ('Number of maximum number tasks running in parallel. You '
+                    'can disable parallel execution by setting this to 1. '
+                    'Defaults to the amount of CPUs on this machine.')
+        settings.register_option(section=section,
+                                 key='max_parallel_tasks',
+                                 default=multiprocessing.cpu_count(),
+                                 key_type=int,
+                                 help_msg=help_msg)
+
+        help_msg = ("Spawn tasks in a specific spawner. Available spawners: "
+                    "'process' and 'podman'")
```

Comment:
We need to improve this later to get the spawners available automatically.

Reply:
Same point here, I just commented something similar below.